source-control/Bug_Reporting_TFS_2010.md

```diff
@@ -0,0 +1,520 @@
+
+The next options have been found useful when a release is composed of several applications and services and a central management of the release is required. These options allow you to:
+
+* Centrally manage the "Iterations"
+* Centrally manage the "Areas" to represent the application/components and services
+* Centrally manage Work Items, Queries and Reports
+* Provide reports at a Product Level
+* Re-assign bugs to different applications/development teams
+* Link Work Items together
+* Promote Code re-use
+
+#### Option 2: Merge into 1 Team Project
+
+* If the multiple applications/components/services are all under the same product or suite of products, they should be contained within 1 Team Project.
+* All aspects (WIs, source code, areas, iterations) can all be managed centrally
+* example:
+  * The MSCBA suite of products, is built and managed by 3 separate development teams
+  * It has 1 Team Project "MSCBA-MDESC" (and all 3 dev teams have access)
+  * It contains:
+    * The Public Facing web site
+    * The Internal web site for Agents
+    * Several web services
+    * and the PL/SQL
+
+#### Option 3: Create a Central Team Project
+
+* If Option 2 is not possible, a new Central Team Project can be created to manage the common functions of the product.
+* All members of each team are granted access to this Central Team Project
+* Source Code continues to remain and be managed within the individual Team Projects
+* Developers are able to check in code against Work Items found in this separate Central Team Project
+* example of "Areas"
+  * EI Processing
+  * ACP
+  * SSA
+  * BenPay
+  * FTS
+  * etc...
+* All bugs are reported in the Central Team Project (and not the individual Team Projects)
+* Bugs can be easily assigned and/or transferred to the other dev teams
+* If a bug touches multiple systems, the original bug becomes the parent bug and new bugs, linked to the parent bug, are created for each impacted system. Only once all child bugs are closed can the parent bug can be closed.
+* The Dev Coordinator and TST Coordinator play a more crucial role in analyzing and filtering the bugs and ensuring the fields are properly set and the bug is assigned to the proper team for resolution and/or for validation.
+
+### Template for "Details = Steps to Reproduce" Field
+
+Below is a template that should be used to populate the field "Steps to Reproduce" under the "Details" Tab.
+
+   **PROBLEM**
+   A short summary of the bug (i.e., 1 or 2 sentences)
+
+   **URL**: URL where the issue was found
+
+   **Environment**: Environment where the issue was found (TST, UAT...)
+   **Date of Occurrence:** If different then the date the bug was created
+   **USER:** Include user that was used when this bug was found and, if pertinent; include information on user role and privileges.  Remove this section
+   if not applicable.
+   **Test Script:** If applicable list the automated test used to identify or reproduce the bug
```

Comment:
Not Blocking: These are being bundled into a single paragraph, I'm not sure if that was intend or not.

Reply:
As intended.